Undo duplicate arakoon configs

sjovdnbos opened the pull request.

Commit:
[wip] apply first changes on vPool creation
todo: look for usage of configs

Commit:
added more (overlooked) files

Commit:
Working change of paths for the vpools.

todo:
remove changes in the vpool metadata
test

Commit:
Removed changes in vpool metadata

todo: further testing
finish migration code

Commit:
changed some constants paths, added migration code
todo:
test migration code

Commit:
Reworked migration code

Commit:
Uncomment Configuration.set

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +40,41 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+            cfg = ArakoonClusterConfig.convert_config_to(cfg, return_type='INI')
+        return cfg
+
+    @staticmethod
```
Use classmethod to avoid having to point towards VPoolShared

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +40,41 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+        return cfg
+
+    @staticmethod
+    def sync_alba_arakoon_config(alba_backend_guid, ovs_client):
```
Does more than just syncing.
It should be retrieve_sync really

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +40,41 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+            Configuration.set(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_backend_guid), remote_config)
+
+            ini_config = ArakoonClusterConfig.convert_config_to(config=remote_config, return_type='INI')
+            Configuration.set(REMOTE_CONFIG_BACKEND_INI.format(alba_backend_guid), ini_config, raw=True)
```
Use a sequence here. That way you guarantee that both entries are updated at the same time and both would fail in case of failures

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,57 @@ def _update_manifest_cache_size(_proxy_config_key):
+            backend_info = metadata['backend']['backend_info']
+            main_backend_guid = backend_info['alba_backend_guid']
+            connection_info = backend_info['connection_info']
+            client = OVSClient(ip=connection_info['host'], port=connection_info['port'], credentials=(connection_info['client_id'],
```
This part is rather volatile. It'd be better to wrap inside try excepts

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,57 @@ def _update_manifest_cache_size(_proxy_config_key):
+                            main_proxy_config[cache_type][1]['albamgr_cfg_url'] = Configuration.get_configuration_path(REMOTE_CONFIG_BACKEND_CONFIG.format(alba_guid))
+
+                    # Set updated config
+                    print 'Setting config at {0}'.format(main_proxy_config_path)
```
print should make way for logging

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,57 @@ def _update_manifest_cache_size(_proxy_config_key):
+
+        ####################################################
+        # deduplicate configs of proxies: now one config per remote alba-backend
+        present_remote_configs = list(Configuration.list('/ovs/framework/remote_configs/alba_backends'))
```
Whole block should be within try excepts

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,57 @@ def _update_manifest_cache_size(_proxy_config_key):
+                            # Make sure replacing config is present
+                            alba_guid = backend_info['alba_backend_guid']
+                            connection_info = backend_info['connection_info']
+                            client = OVSClient(ip=connection_info['host'], port=connection_info['port'], credentials=(connection_info['client_id'], connection_info['client_secret']))
```
`OVSClient.get_instance(connection_info=connection_info, cache_store=VolatileFactory.get_client())`

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,57 @@ def _update_manifest_cache_size(_proxy_config_key):
+                            config_to_delete = os.path.join(proxy_config_template, other_cache_config)
+                            print 'deleting {0}'.format(config_to_delete)
+                            Configuration.delete(config_to_delete)
+
```
Nothing here to indicate that this migration has been done

Commit:
Fixed remarks

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,84 @@ def _update_manifest_cache_size(_proxy_config_key):
+
+                            # Now remove old configs of proxies
+                            other_cache_configs = list(Configuration.list(proxy_config_template))
+                            if not other_cache_configs or proxy_errors:
```
if other_cache_configs and not proxy_errors:
 ....
Don't do if: pass

Commit:
Extensions:
-Implemented 'Extract_key_from_path'

Framework: End-of-day-commit
-Split up 'retrieve' and 'sync' config functionality
-Fixed issue in migration code where proxy cacc's were incorrect
-todo: fix sync: fix leak in configmgmt

Commit:
Removed check for usage of backend. This check wouldnt allow creation of vpools with remote backends

Commit:
Extensions: removed obsolete letter from contstants name
framework: added logic to migration code and fixed api call usage

JeffreyDevloo commented on ovs/lib/helpers/storagedriver/installer.py:
```diff
@@ -340,12 +341,15 @@ def setup_proxy_configs(self):
-
             if cache_settings['is_backend'] is True:
-                cfg_tree_name = 'abm_bc' if cache_type == StorageDriverConfiguration.CACHE_BLOCK else 'abm_aa'
+                if cache_type == StorageDriverConfiguration.CACHE_BLOCK:
```
Required to fetch the alba_backend_guid like that? Seems like a check that isnt required

sjovdnbos replied:
Valid point, i am removing the check. If for some reason, the cache type is an invalid string, it will raise as the key will not exist

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+        :return: Arakoon configuration information
+        :rtype: dict
+        """
+        #todo: Fix leak of backends not bein removed
```
`being`

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+
+    @classmethod
+    def calculate_abm_configs_in_use(cls):
+        """
```
typing

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+        cache_types = [StorageDriverConfiguration.CACHE_FRAGMENT, StorageDriverConfiguration.CACHE_BLOCK]
+
+        for vpool in VPoolList.get_vpools():
+            print vpool.name
```
Leftover print

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+
+        for vpool in VPoolList.get_vpools():
+            print vpool.name
+            proxy_config_template = '/ovs/vpools/{0}/proxies/{{0}}/config/main'.format(vpool.guid)
```
Can we keep this constant in a constants file?

sjovdnbos replied:
https://github.com/openvstorage/framework-extensions/pull/194

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+                    cfg_mgr_url = cfg['albamgr_cfg_url']  # type:str
+
+                    if cfg_mgr_url not in present_remote_configs.values():
+                        print cfg_mgr_url
```
Stdout is my logger!

JeffreyDevloo commented on ovs/lib/helpers/vpool/shared.py:
```diff
@@ -38,3 +43,79 @@ def retrieve_alba_arakoon_config(alba_backend_guid, ovs_client):
+                        in_use.add(Configuration.extract_key_from_path(cfg_mgr_url))
+                    # Extract all albamgr_cfg_urls
+                    for cache_type in cache_types:
+                        cache_cfg = cfg.get(cache_type)
```
If you want to ensure that you get a list, use .get(key, [])
len(None) -> error

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -307,7 +312,7 @@ def _update_metadata_structure(metadata):
                                 new_cache_structure['is_backend'] = True
                                 # Copy over the old data
-                                new_cache_structure['backend_info']['arakoon_config'] = backend_data['arakoon_config']
+                                new_cache_structure['backend_info']['arakoon_config'] = backend_data['arakoon_config'] #todo fix migration code
```
Todo or not todo

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,90 @@ def _update_manifest_cache_size(_proxy_config_key):
+        # deduplicate configs of proxies: now one config per remote alba-backend
+        present_remote_configs = list(Configuration.list(REMOTE_CONFIG_BACKEND_BASE))
+        errors = []
+        #todo add check if needs to migrate
```
That is the question

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,90 @@ def _update_manifest_cache_size(_proxy_config_key):
         MigrationController._logger.info('Finished out of band migrations')
+
+        ####################################################
+        # deduplicate configs of proxies: now one config per remote alba-backend
```
since when do we write our comments without a capital letter?
https://i.kym-cdn.com/photos/images/facebook/001/209/105/7b2.jpeg

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +669,90 @@ def _update_manifest_cache_size(_proxy_config_key):
+                                        errors.append('Failed to update proxy {0} of vPool {1}: {2}'.format(proxy.guid, vpool.guid, ex))
+                                        continue
+                            # Set updated config.
+                            Configuration.set(main_proxy_config_path, main_proxy_config)
```
Set and delete can be in a delete

Commit:
Apply vpools constants changes throughout our codebase

Commit:
Apply missing changes

Commit:
Fixed overlooked format

Commit:
Now properly checks for containing a string

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +670,92 @@ def _update_manifest_cache_size(_proxy_config_key):
+
+        ####################################################
+        # Deduplicate configs of proxies: now one config per remote alba-backend
+        present_remote_configs = list(Configuration.list(REMOTE_CONFIG_BACKEND_BASE))
```
Move in the try

JeffreyDevloo commented on ovs/lib/migration.py:
```diff
@@ -664,3 +670,92 @@ def _update_manifest_cache_size(_proxy_config_key):
+                                for other_cache_config in other_cache_configs:
+                                    config_to_delete = os.path.join(config_path, other_cache_config)
+                                    Configuration.delete(config_to_delete, transaction=transaction_id)
+                            else:
```
Remove else and move comment above the if

Commit:
Fix remarks
